src/main/java/com/exactpro/th2/readcsv/impl/ProtoReaderFactory.java

```diff
@@ -0,0 +1,139 @@
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.readcsv.impl;
+
+import com.exactpro.th2.common.grpc.EventBatch;
+import com.exactpro.th2.common.grpc.EventID;
+import com.exactpro.th2.common.grpc.RawMessage;
+import com.exactpro.th2.common.grpc.RawMessageBatch;
+import com.exactpro.th2.common.message.MessageUtils;
+import com.exactpro.th2.common.schema.factory.CommonFactory;
+import com.exactpro.th2.common.schema.message.MessageRouter;
+import com.exactpro.th2.read.file.common.AbstractFileReader;
+import com.exactpro.th2.read.file.common.DirectoryChecker;
+import com.exactpro.th2.read.file.common.MovedFileTracker;
+import com.exactpro.th2.read.file.common.StreamId;
+import com.exactpro.th2.read.file.common.impl.ProtoDefaultFileReader;
+import com.exactpro.th2.read.file.common.state.impl.InMemoryReaderState;
+import com.exactpro.th2.readcsv.cfg.ReaderConfig;
+import com.google.protobuf.ByteString;
+import kotlin.Unit;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.LineNumberReader;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ProtoReaderFactory extends ReaderAbstractFactory {
+
+    public ProtoReaderFactory(ReaderConfig configuration, CommonFactory commonFactory) {
+        super(configuration, commonFactory);
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProtoReaderFactory.class);
+
+    @Override
+    protected AbstractFileReader<LineNumberReader, ?, ?> buildReader(MessageRouter<EventBatch> eventBatchRouter, DirectoryChecker directoryChecker, EventID rootId) {
+        var headerHolder = new ProtoHeaderHolder(configuration.getAliases());
+        var router = commonFactory.getMessageRouterRawBatch();
+
+        return new ProtoDefaultFileReader.Builder<>(
+                configuration.getCommon(),
+                directoryChecker,
+                new ProtoCsvContentParser(configuration.getAliases()),
+                new MovedFileTracker(configuration.getSourceDirectory()),
+                new InMemoryReaderState(),
+                streamId -> commonFactory.newMessageIDBuilder().build(),
+                ReaderAbstractFactory::createSource
+        )
+                .readFileImmediately()
+                .acceptNewerFiles()
+                .onSourceFound((streamId, path) -> clearHeader(headerHolder, streamId))
+                .onContentRead((streamId, path, builders) -> attachHeaderOrHold(headerHolder, streamId, builders, configuration))
+                .onStreamData((streamId, builders) -> publishMessages(router, streamId, builders))
+                .onError((streamId, message, ex) -> publishErrorEvent(eventBatchRouter, streamId, message, ex, rootId))
+                .onSourceCorrupted((streamId, path, e) -> publishSourceCorruptedEvent(eventBatchRouter, path, streamId, e, rootId))
```

Comment:
Part of those method invocations can be moved to the abstract class I think (readFileImmediately, acceptNewerFiles,onError, onSourceCorrupted)